src/qililab/result/counts.py

```diff
@@ -124,3 +126,56 @@ def __iadd__(self, other: Counts) -> Counts:
         """
         self.add_counts(other)
         return self
+
+    @staticmethod
+    def from_qprogram(qprogram_results, qubit_mapping: list[str] | None = None):
+        """Returns `Count` object from a `QProgramResults`.
+
+        Args:
+            qprogram_results (QProgramResults): The QProgramResults object we want to get the `Counts` from.
+            qubit_mapping (list[str], optional): A list containing the name of the busses to map.
+                The buses are map to qubits on the same index, a bus at the i-th element in the list is mapped to the i-th qubit.
+                Defaults to None.
+
+        Raises:
+            ValueError: When the `results` attribute from `qprogram_results` is an empty dictionary.
+            ValueError: When a qubit mapping is incomplete and does not map all qubits.
+            ValueError: When a qubit mapping is specified and any of the busses does not match with any on the runcard.
+
+        Returns:
+            dict[str, float]: Dictionary containing the quantum states as the keys of the dictionary, and the
+                probabilities obtained for each state as the values of the dictionary.
+        """
+        if not qprogram_results.results:
+            raise ValueError(f"Can not obtain counts with no measurments, {qprogram_results.__class__} empty")
+
+        n_qubits = len(qprogram_results.results)
+        counts_object = Counts(n_qubits)
+        buses = list(qprogram_results.results.keys())
+
+        if qubit_mapping is not None:
+            unique_mapping = set(qubit_mapping)  # Remove possible repeated elements
+            if n_qubits != len(unique_mapping):
+                raise ValueError(
+                    f"Expected mapping for all qubits. Results have {n_qubits} qubits, but only {len(unique_mapping)} diferent buses were specified on the mapping."
+                )
+            if not unique_mapping.issubset(set(buses)):
+                raise ValueError(
+                    "No measurements found for all specified buses, check the name of the buses provided with the mapping match all the buses specified in runcard."
+                )
+            buses = qubit_mapping
+
+        # The threshold inside of a qblox bin is the name they use for already classified data as a value between
+        # 0 and 1, not the value used in the comparator to perform such classification.
+        th_matrix = np.array(
```

Comment:
Can we avoid abbreviations? We can rename this to `threshold_matrix`

Reply:
Done!